contracts/mainnet/connectors/compound-iii/main.sol

```diff
@@ -0,0 +1,907 @@
+	 * @param minCollateralAmt Minimum amount of collateral expected to be received.
+	 * @param baseAmt Amount of base asset to be sold for collateral.
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of base tokens sold.
+	 */
+	function buyCollateral(
+		address market,
+		address asset,
+		address dest,
+		uint256 minCollateralAmt,
+		uint256 baseAmt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, baseAmt);
+
+		bool isEth = asset == ethAddr;
+		address token_ = isEth ? wethAddr : asset;
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		if (isEth) {
+			convertEthToWeth(isEth, tokenContract, amt_);
+		}
+
+		CometInterface(market).buyCollateral(
+			asset,
+			minCollateralAmt,
+			amt_,
+			dest
+		);
+
+		uint256 collAmt = CometInterface(market).quoteCollateral(asset, amt_);
+		setUint(setId, amt_);
+
+		eventName_ = "LogBuyCollateral(address,address,uint256,uint256,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(
+			market,
+			asset,
+			baseAmt,
+			minCollateralAmt,
+			collAmt,
+			getId,
+			setId
+		);
+	}
+
+	/**
+	 * @dev Claim rewards and interests accrued in supplied/borrowed base asset.
+	 * @notice Claim rewards and interests accrued.
+	 * @param market The address of the market.
+	 * @param account The account of which the rewards are to be claimed.
+	 * @param accrue Should accrue the rewards and interest before claiming.
+	 * @param setId ID stores the amount of rewards claimed.
+	 */
+	function claimRewards(
+		address market,
+		address account,
+		bool accrue,
+		uint256 setId
+	) public returns (string memory eventName_, bytes memory eventParam_) {
+		cometRewards.claim(market, account, accrue);
+
+		//in reward token decimals
+		uint256 totalRewardsClaimed = cometRewards.rewardsClaimed(
+			market,
+			account
+		);
+		setUint(setId, totalRewardsClaimed);
+
+		eventName_ = "LogRewardsClaimed(address,address,uint256,uint256,bool)";
+		eventParam_ = abi.encode(
+			market,
+			account,
+			totalRewardsClaimed,
+			setId,
+			accrue
+		);
+	}
+
+	/**
+	 * @dev Claim rewards and interests accrued in supplied/borrowed base asset.
+	 * @notice Claim rewards and interests accrued and transfer to dest address.
+	 * @param market The address of the market.
+	 * @param account The account of which the rewards are to be claimed.
+	 * @param dest The account where to transfer the claimed rewards.
+	 * @param accrue Should accrue the rewards and interest before claiming.
+	 * @param setId ID stores the amount of rewards claimed.
+	 */
+	function claimRewardsTo(
+		address market,
+		address account,
+		address dest,
+		bool accrue,
+		uint256 setId
+	) public returns (string memory eventName_, bytes memory eventParam_) {
+		cometRewards.claimTo(market, account, dest, accrue);
+
+		//in reward token decimals
+		uint256 totalRewardsClaimed = cometRewards.rewardsClaimed(
```

Comment:
@Richa-iitr This returns all the rewards claimed from beginning, right? But we need the rewards claimed in that transaction.

Reply:
changed to owed amount.